helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -94,29 +95,47 @@ public BestPossibleExternalViewVerifier(HelixZkClient zkClient, String clusterNa
     _dataProvider = new ResourceControllerDataProvider();
   }
 
-  public static class Builder {
-    private String _clusterName;
+  private BestPossibleExternalViewVerifier(Builder builder) {
+    super(builder);
+    // Deep copy data from Builder
```

Comment:
It seems the original constructors don't do deep copy. Is there a reason we do deep copy here? Maybe to prevent it from any concurrent modification exception as seen in prod or tests?

Reply:
This is because users can re-use a builder object to create multiple instances.